packages/eventual-send/src/local.js

```diff
@@ -1,38 +1,49 @@
-const q = JSON.stringify;
-
 const ntypeof = specimen => (specimen === null ? 'null' : typeof specimen);
 
 export const localApplyFunction = (t, args) => {
-  if (!(t instanceof Function)) {
-    const ftype = ntypeof(t);
-    throw TypeError(
-      `Cannot invoke target as a function; typeof target is ${q(ftype)}`,
-    );
-  }
+  const { details: X, quote: q } = assert;
+  assert.typeof(
+    t,
+    'function',
+    X`Cannot invoke target as a function; typeof target is ${q(ntypeof(t))}`,
+    TypeError,
+  );
   return t(...args);
 };
 
 export const localApplyMethod = (t, method, args) => {
+  const { details: X, quote: q } = assert;
   if (method === undefined || method === null) {
+    // Base case; bottom out to apply functions.
     return localApplyFunction(t, args);
   }
   if (t === undefined || t === null) {
-    const ftype = ntypeof(t);
-    throw TypeError(
-      `Cannot deliver ${q(method)} to target; typeof target is ${q(ftype)}`,
+    assert.fail(
+      X`Cannot deliver ${q(method)} to target; typeof target is ${q(
+        ntypeof(t),
+      )}`,
+      TypeError,
     );
   }
   const fn = t[method];
-  if (!(fn instanceof Function)) {
-    const ftype = ntypeof(fn);
-    if (ftype === 'undefined') {
-      const names = Object.getOwnPropertyNames(t).sort();
-      throw TypeError(`target has no method ${q(method)}, has ${q(names)}`);
-    }
-    throw TypeError(
-      `invoked method ${q(method)} is not a function; it is a ${q(ftype)}`,
+  if (fn === undefined) {
+    // Reflect.ownKeys only works on objects, so we use the keys of the
+    // ownPropertyDescriptors object.
+    const skeys = Reflect.ownKeys(Object.getOwnPropertyDescriptors(t))
+      .map(k => String(k))
```

Comment:
I'm ok either way, but is there a reason you didn't do

```suggestion
      .map(String)
```
?

Reply:
Defensively, I don't want to supply the array index as the second argument to `String`.